src/v/redpanda/admin/debug.cc

```diff
@@ -810,6 +817,79 @@ admin_server::get_partition_state_handler(
     co_return ss::json::json_return_type(std::move(response));
 }
 
+ss::future<ss::json::json_return_type>
+admin_server::get_producers_state_handler(
+  std::unique_ptr<ss::http::request> req) {
+    const model::ntp ntp = parse_ntp_from_request(req->param);
+    auto timeout = std::chrono::duration_cast<model::timeout_clock::duration>(
+      10s);
+    auto result = co_await _tx_gateway_frontend.local().get_producers(
+      cluster::get_producers_request{ntp, timeout});
+    if (result.error_code != cluster::tx::errc::none) {
+        throw ss::httpd::server_error_exception(fmt::format(
+          "Error {} processing partition state for ntp: {}",
```

Comment:
nit: "processing partition state" sounds a bit ambiguous, maybe "getting producers for ntp:" instead?